pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ExplainIntegrationTestTrait.java

```
@@ -0,0 +1,124 @@
+
+      Assert.assertEquals(plan.asText(), expected);
+    } catch (RuntimeException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  default void explainSse(boolean verbose, @Language("sql") String query, Object... expected) {
+    try {
+      String actualQuery = "SET useMultistageEngine=false; explain plan for " + query;
+      if (verbose) {
+        actualQuery = "SET explainPlanVerbose=true; " + actualQuery;
+      }
+      JsonNode jsonNode = postQuery(actualQuery);
+      JsonNode plan = jsonNode.get("resultTable").get("rows");
+      List<String> planAsStrList = (List<String>) JsonUtils.jsonNodeToObject(plan, List.class).stream()
+          .map(Object::toString)
+          .collect(Collectors.toList());
+
+      for (int i = 0; i < planAsStrList.size(); i++) {
+        String planAsStr = planAsStrList.get(i);
+        if (i < expected.length) {
+          Object expectedObj = expected[i];
+          if (expectedObj instanceof Pattern) {
+            Assert.assertTrue(((Pattern) expectedObj).matcher(planAsStr).matches(),
+                "Actual: " + planAsStr + ", Expected: " + expectedObj);
+          } else if (expectedObj instanceof String) {
+            Assert.assertEquals(planAsStr, expectedObj, "Actual: " + planAsStr + ", Expected: " + expectedObj);
+          } else {
+            Assert.fail("Expected object should be either Pattern or String");
+          }
+        } else {
+          Assert.fail("Expected: " + expected.length + " elements, but more were found. Remaining: "
+              + planAsStrList.subList(i, planAsStrList.size()));
+        }
+      }
+      if (planAsStrList.size() < expected.length) {
+        Assert.fail("Expected: " + expected.length + " elements, but only " + planAsStrList.size() + " were found. "
+            + "Missing: " + Arrays.asList(expected).subList(planAsStrList.size(), expected.length));
+      }
```

nit: could these checks be moved to before the loop execution to fail fast?

---

I don't think this is going to be problematic given this is going to iterate over lines, which are going to be in the order of few dozens at most. But I'm changing it anyway and adding a bit of extra context.